dragonfly/exd/exd_core.py

```diff
@@ -509,7 +520,7 @@ def _dispatch_single_experiment_to_worker_manager(self, qinfo):
     # Create a new qinfo namespace and dispatch new job.
     qinfo.send_time = self.get_curr_spent_capital()
     qinfo.step_idx = self.step_idx
-    self.worker_manager.dispatch_single_experiment(self.experiment_caller, qinfo)
+    self.worker_manager.dispatch_single_experiment(self.experiment_caller, qinfo, **{"ask_tell_mode": self.ask_tell_mode})
```

Comment:
Why do we need this in ask-tell mode? In fact, there is no need for a worker manager in ask-tell mode

Reply:
When dispatching an experiment, there is a point where the qinfo needs to be evaluated via `func_caller.eval_from_qinfo()`, and keyword-arguments are already passed in as one of its arguments. However, with no function in the function caller, there is no way to evaluate the point, so within the `eval_from_qinfo` we just return the qinfo if the optimizer is in ask-tell mode, and this information can only be passed in as an extra argument. 

The worker manager is set to a default if the user doesn't specify one.

Reply:
Correct me if I'm wrong - @kirthevasank's comment is specifically referencing `ask_tell` mode. And in this mode, you never call "dispatch_experiment". Right?

Reply:
Technically, `dispatch_experiment` is called, but the routine within it is slightly different in `ask_tell_mode`. Basically, since the `FunctionCaller`'s `func` is None, there is no evaluation for that point (done in `[eval_from_qinfo]`(https://github.com/dragonfly/dragonfly/pull/53/files/e7408731a419a20e84b54bfa04662ebb3e97e73d#diff-ec1a0cf496cb486f90c712f876345fa5R222)). However, we need to load in things like `eval_time` and `receive_time` for booking keeping into the qinfo to ensure other functionality remains the same.

Reply:
Ok got it. Can you abstract that functionality out into a helper function and avoid using the worker manager all together? It seems like bookkeeping is not critical to belonging to the worker manager.

Reply:
Meaning avoid the worker manager in asktell mode.

Reply:
Updated, please see changes in new commit